clang/lib/Basic/Attributes.cpp

```diff
@@ -8,13 +8,14 @@ using namespace clang;
 int clang::hasAttribute(AttrSyntax Syntax, const IdentifierInfo *Scope,
                         const IdentifierInfo *Attr, const TargetInfo &Target,
                         const LangOptions &LangOpts) {
+  StringRef ScopeName = Scope ? Scope->getName() : "";
   StringRef Name = Attr->getName();
   // Normalize the attribute name, __foo__ becomes foo.
-  if (Name.size() >= 4 && Name.startswith("__") && Name.endswith("__"))
+  if (ScopeName != "__sycl_detail__" && Name.size() >= 4 &&
```

Comment:
Just for my education, did I understand it correctly that both forms `[[sycl_detail::uses_aspects()]]` and `[[__sycl_detail__::__uses_aspects__()]]` are accepted?

Reply:
Right now no.  Prior to making this change, I expected both to work. But for some reason __uses_aspects__ just did not work. I tried debugging it for sometime to see why it was not getting normalized correctly but could not figure it out. I then noticed that this file has many other functions doing normalization and it is only done for gnu and clang namespace. So I added a guard in this function to not normalize for `__sycl_detail__ ` namespace.

Reply:
I added a FIXME here because I am curious about why this is not working. Since the specification only requires `__uses_aspects__` to work I think adding this guard for now is fine, but for upstreaming we may need to figure this out at some point.

Reply:
Got that, thanks. I'm not sure we even need "uses_aspects" to be supported, since the design doc (github.com/intel/llvm/blob/sycl/sycl/doc/OptionalDeviceFeatures.md) says that it should be spelled with double underscores in DPC++ headers anyway. Just asked for better understanding.
I would assume then that to make both forms work we should declare the attribute in `Attr.td` without underscores and then its name will be normalized somewhere in this file.

Reply:
I tried that but it didn't work. Not sure if I made some unintentional mistake though